docs/source/depth_estimation.mdx

```diff
@@ -0,0 +1,198 @@
+```
+
+**Note** that the above utility is taken from the [FastDepth repository](https://github.com/dwofk/fast-depth/blob/master/utils.py). You can now visualize the depth map.
+
+```py 
+>>> depth_target = train_dataset[index]["depth_map"]
+>>> depth_target = np.squeeze(np.array(depth_target))
+
+>>> d_min = np.min(depth_target)
+>>> d_max = np.max(depth_target)
+>>> depth_target = colored_depthmap(depth_target, d_min, d_max)
+
+>>> plt.imshow(depth_target.astype("uint8"))
+>>> plt.axis("off")
+>>> plt.show()
+```
+
+<div class="flex justify-center">
+    <img src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/datasets/depth_est_target_viz.png">
+</div>
+
+Now apply some augmentations with `albumentations`. The augmentation transformations include:
+
+* Random horizontal flipping
+* Random cropping 
+* Random brightness and contrast 
+* Random gamma correction 
+* Random hue saturation
+
+```py 
+>>> import albumentations as A
+
+>>> crop_size = (448, 576)
+>>> transforms = [
+...     A.HorizontalFlip(),
+...     A.RandomCrop(crop_size[0], crop_size[1]),
+...     A.RandomBrightnessContrast(),
+...     A.RandomGamma(),
+...     A.HueSaturationValue()
+... ]
+```
+
+Since these transformations (such as flipping, cropping) also impact the depth maps, you need to define a mapping
+while applying these transformations:
+
+```py 
+>>> additional_targets = {"depth": "mask"}
+>>> aug = A.Compose(transforms=transforms, additional_targets=additional_targets)
+```
+
+This will ensure that you can pass an image and its corresponding depth map to `aug`.
+
+Create a function to apply the transformation to the images as well as their depth maps:
+
+```py 
+>>> def transforms(examples):
+...     transformed_images, transformed_maps = [], []
+...     for image, depth_map in zip(examples["image"], examples["depth_map"]):
+...         image, depth_map = np.array(image), np.array(depth_map)
+...         transformed = aug(image=image, depth=depth_map)
+...         transformed_images.append(transformed["image"])
+...         transformed_maps.append(transformed["depth"])
+...
+...     examples["pixel_values"] = transformed_images
+...     examples["labels"] = transformed_maps
+...     return examples
+```
+
+Use the [`~Dataset.set_transform`] function to apply the transformation on-the-fly to batches of the dataset to consume less disk space:
+
+```py
+>>> train_dataset.set_transform(transforms)
+```
+
+You can verify the transformation worked by indexing into the `pixel_values` and `labels` of an example:
+
+```py
+>>> image = np.array(train_dataset[index]["pixel_values"])
+>>> depth_map = np.array(train_dataset[index]["labels"])
+```
+
+```py
+>>> plt.imshow(image)
+>>> plt.axis("off")
+>>> plt.show()
+```
+
+<div class="flex justify-center">
+    <img src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/datasets/depth_est_sample_aug.png">
+</div>
+
+```py 
```

Comment:
Maybe add some text here that says something like:

`Visualize the same transformation on the image's corresponding depth map:`